packages/cinterop/src/jvmMain/CMakeLists.txt

```diff
@@ -4,18 +4,40 @@ find_package(JNI)
 set(SWIG_JNI_GENERATED "${CMAKE_SOURCE_DIR}/../../../jni-swig-stub/build/generated/sources/jni")
 set(SWIG_JNI_HELPERS "${CMAKE_SOURCE_DIR}/../../../jni-swig-stub/src/main/jni")
 set(CINTEROP_JNI "${CMAKE_SOURCE_DIR}/../jvm/jni")
+set(REALM_TARGET_LINK_LIBS RealmFFIStatic Realm::ObjectStore)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")
+if (CMAKE_SYSTEM_NAME MATCHES "^Windows")
+    MESSAGE("Building JNI for Windows")
+    set(CMAKE_CXX_STANDARD 17)
+    set(CMAKE_CXX_STANDARD_REQUIRED ON)
+    set(REALM_INCLUDE_DIRS "${CMAKE_BINARY_DIR}/src" "${JAVA_INCLUDE_PATH}" "${JAVA_INCLUDE_PATH}/win32" "${CINTEROP_JNI}" "${SWIG_JNI_GENERATED}" "${SWIG_JNI_HELPERS}")
 
+elseif (CMAKE_SYSTEM_NAME MATCHES "^Android")
+    MESSAGE("Building JNI for Android")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")
+    set(CAPI_BUILD "${CMAKE_SOURCE_DIR}/../../../external/core}/build-android-${ANDROID_ABI}-${CMAKE_BUILD_TYPE}")
+    set(REALM_INCLUDE_DIRS ${CAPI_BUILD}/src ${CINTEROP_JNI} ${SWIG_JNI_GENERATED} ${SWIG_JNI_HELPERS})
+    set(REALM_TARGET_LINK_LIBS log android RealmFFIStatic Realm::ObjectStore)
```

Comment:
It feels a bit akward that we also define `RealmFFIStatic Realm::ObjectStore`.... would it instead make sense to  rename `REALM_TARGET_LINK_LIBS` to `SHARED_REALM_TARGET_LINK_LIBS` (or something) and then set `REALM_TARGET_LINK_LIBS` for each target?

Reply:
Splitting this variable will force us to set the target for each platform, whereas now only Android is the exception (the other 3 inherit the default linking behaviour)